host/src/types/gatt_traits.rs

```diff
@@ -138,22 +143,38 @@ impl<const N: usize> GattValue for [u8; N] {
             data.try_into().map_err(|_| FromGattError::InvalidLength)
         }
     }
+}
+
+impl<const N: usize> ToGatt for [u8; N] {
+    const MIN_SIZE: usize = 0;
+    const MAX_SIZE: usize = N;
 
     fn to_gatt(&self) -> &[u8] {
         self.as_slice()
     }
 }
 
-impl<const N: usize> GattValue for String<N> {
-    const MIN_SIZE: usize = 0;
-    const MAX_SIZE: usize = N;
-
+impl<const N: usize> FromGatt for String<N> {
     fn from_gatt(data: &[u8]) -> Result<Self, FromGattError> {
         String::from_utf8(unwrap!(Vec::from_slice(data).map_err(|_| FromGattError::InvalidLength)))
             .map_err(|_| FromGattError::InvalidCharacter)
     }
+}
+
+impl<const N: usize> ToGatt for String<N> {
+    const MIN_SIZE: usize = 0;
+    const MAX_SIZE: usize = N;
 
     fn to_gatt(&self) -> &[u8] {
         self.as_ref()
     }
 }
+
+impl ToGatt for &'static str {
+    const MIN_SIZE: usize = 0;
+    const MAX_SIZE: usize = usize::MAX;
+
+    fn to_gatt(&self) -> &[u8] {
+        self.as_bytes()
+    }
+}
```

Comment:
Does this have to be static? I'd imagine almost any use case would have it be static but I wonder if you can get a little extra flexibility by using a generic lifetime. This should work the same for a &`static str I believe while also allowing non-static `&str`s to be used assuming the borrow checker agrees.

```suggestion
impl<'a> ToGatt for &'a str {
    const MIN_SIZE: usize = 0;
    const MAX_SIZE: usize = usize::MAX;

    fn to_gatt(&self) -> &'a [u8] {
        self.as_bytes()
    }
}
```

Reply:
See the discussion here https://github.com/embassy-rs/trouble/issues/272 @jscatena88

Reply:
It looks like the complications there were from implementing `from_gatt` not `to_gatt`. I'm still somewhat confident that the code I wrote above should work actually. It is passing tests locally on my laptop (except for one test that seems to be expecting connected hardware? `gatt_client_server`). But either way not a big deal